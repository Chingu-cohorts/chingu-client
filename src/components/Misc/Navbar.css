@media (min-width: 600px) {
  nav .brand-logo {
    margin-left: 2rem;
  }
}
