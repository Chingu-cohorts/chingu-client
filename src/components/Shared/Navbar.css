.nav-header {
  width: 100%;
  background: #ffffff;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid #e9e9e9;
  padding: 0;
}

.nav-header .logo {
  width: 120px;
  height: 31px;
  font-size: 2rem;
  font-weight: 600;
}

.nav-header .logo .logo-container {
  margin-left: 1rem;
}

.nav-header .ant-menu-horizontal {
  border-bottom: none;
}
