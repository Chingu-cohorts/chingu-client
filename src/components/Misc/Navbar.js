import React, { Component } from "react";
import { Link } from "react-router-dom";

import "./Navbar.css";

class Navbar extends Component {
  handleLogout(e) {
    e.preventDefault();

    window.localStorage.removeItem("token");
    window.location = "/";
  }

  render() {
    return (
      <nav className="green accent-3">
        <div className="nav-wrapper">
          <Link to="/" className="brand-logo">
            친구
          </Link>
          <ul id="nav-mobile" className="right hide-on-med-and-down">
            <li>
              <Link to="/profile/2">Profile</Link>
            </li>
            <li>
              <Link to="/apply">Apply</Link>
            </li>
            <li>
              <Link to="/login">Login</Link>
            </li>
            <li>
              <a onClick={e => this.handleLogout(e)}>Logout</a>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default Navbar;
