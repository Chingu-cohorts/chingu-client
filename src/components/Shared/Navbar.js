import React, { Component } from "react";
import { graphql } from "react-apollo";
import { Link } from "react-router-dom";
import { Layout, Menu } from "antd";

import currentUserQuery from "./currentUserQuery";
import "./Navbar.css";

const { Header } = Layout;

export class Navbar extends Component {
  handleLogout = e => {
    e.preventDefault();
    window.localStorage.removeItem("token");
    window.location = "/";
  };

  render() {
    let { loading, user } = this.props.data;
    if (loading) {
      return <h1>Loading</h1>;
    }
    return (
      <Layout>
        <Header className="nav-header">
          <div className="logo">
            <div className="logo-container">
              <Link to="/">Chingu</Link>
            </div>
          </div>
          <Menu
            theme="light"
            mode="horizontal"
            style={{ lineHeight: "64px", display: "flex" }}
          >
            {user ? null : (
              <Menu.Item key="2">
                <Link to="/signup">Sign Up</Link>
              </Menu.Item>
            )}

            {user ? null : (
              <Menu.Item key="3">
                <Link to="/login">Login</Link>
              </Menu.Item>
            )}

            {user ? (
              <Menu.Item key="4">
                <Link to={`/profile/${user.id}`}>Profile</Link>
              </Menu.Item>
            ) : null}

            {user ? (
              <Menu.Item key="5">
                <Link to="/settings">Settings</Link>
              </Menu.Item>
            ) : null}

            {user ? (
              <Menu.Item key="6">
                <a onClick={this.handleLogout}>Logout</a>
              </Menu.Item>
            ) : null}
          </Menu>
        </Header>
      </Layout>
    );
  }
}

export default graphql(currentUserQuery)(Navbar);
