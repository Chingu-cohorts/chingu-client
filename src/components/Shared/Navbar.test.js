import React from "react";
import { shallow } from "enzyme";

import { Navbar } from "./Navbar";

describe("Navbar Component", () => {
  it("renders with text in case the component is loading", () => {
    const props = {
      data: {
        loading: true,
        user: null
      }
    };
    const wrapper = shallow(<Navbar {...props} />);
    const message = <h1>Loading</h1>;
    expect(wrapper.contains(message)).toEqual(true);
  });

  it("renders without crashing", () => {
    const props = {
      data: {
        loading: false,
        user: {
          id: 1
        }
      }
    };
    const wrapper = shallow(<Navbar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
