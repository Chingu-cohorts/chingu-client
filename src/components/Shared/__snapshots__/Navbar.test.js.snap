// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    data={
        Object {
            "loading": false,
            "user": Object {
              "id": 1,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Adapter
        className="nav-header"
>
        <div
                className="logo"
        >
                Chingu
        </div>
        <Menu
                className=""
                mode="horizontal"
                prefixCls="ant-menu"
                style={
                        Object {
                                "lineHeight": "64px",
                              }
                }
                theme="light"
        >
                <MenuItem>
                        <Link
                                replace={false}
                                to="/"
                        >
                                Home
                        </Link>
                </MenuItem>
                <MenuItem>
                        <Link
                                replace={false}
                                to="/profile/1"
                        >
                                Profile
                        </Link>
                </MenuItem>
                <MenuItem>
                        <Link
                                replace={false}
                                to="/settings"
                        >
                                Settings
                        </Link>
                </MenuItem>
                <MenuItem>
                        <a
                                onClick={[Function]}
                        >
                                Logout
                        </a>
                </MenuItem>
        </Menu>
</Adapter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="logo"
>
            Chingu
</div>,
          <Menu
            className=""
            mode="horizontal"
            prefixCls="ant-menu"
            style={
                        Object {
                                    "lineHeight": "64px",
                                  }
            }
            theme="light"
>
            <MenuItem>
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    Home
                        </Link>
            </MenuItem>
            <MenuItem>
                        <Link
                                    replace={false}
                                    to="/profile/1"
                        >
                                    Profile
                        </Link>
            </MenuItem>
            <MenuItem>
                        <Link
                                    replace={false}
                                    to="/settings"
                        >
                                    Settings
                        </Link>
            </MenuItem>
            <MenuItem>
                        <a
                                    onClick={[Function]}
                        >
                                    Logout
                        </a>
            </MenuItem>
</Menu>,
        ],
        "className": "nav-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Chingu",
            "className": "logo",
          },
          "ref": null,
          "rendered": "Chingu",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem>
                <Link
                                replace={false}
                                to="/"
                >
                                Home
                </Link>
</MenuItem>,
              null,
              null,
              <MenuItem>
                <Link
                                replace={false}
                                to="/profile/1"
                >
                                Profile
                </Link>
</MenuItem>,
              <MenuItem>
                <Link
                                replace={false}
                                to="/settings"
                >
                                Settings
                </Link>
</MenuItem>,
              <MenuItem>
                <a
                                onClick={[Function]}
                >
                                Logout
                </a>
</MenuItem>,
            ],
            "className": "",
            "mode": "horizontal",
            "prefixCls": "ant-menu",
            "style": Object {
              "lineHeight": "64px",
            },
            "theme": "light",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
>
                  Home
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": [Function],
            },
            null,
            null,
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/profile/1"
>
                  Profile
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Profile",
                  "replace": false,
                  "to": "/profile/1",
                },
                "ref": null,
                "rendered": "Profile",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/settings"
>
                  Settings
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Settings",
                  "replace": false,
                  "to": "/settings",
                },
                "ref": null,
                "rendered": "Settings",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "class",
              "props": Object {
                "children": <a
                  onClick={[Function]}
>
                  Logout
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Logout",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Logout",
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Adapter
          className="nav-header"
>
          <div
                    className="logo"
          >
                    Chingu
          </div>
          <Menu
                    className=""
                    mode="horizontal"
                    prefixCls="ant-menu"
                    style={
                              Object {
                                        "lineHeight": "64px",
                                      }
                    }
                    theme="light"
          >
                    <MenuItem>
                              <Link
                                        replace={false}
                                        to="/"
                              >
                                        Home
                              </Link>
                    </MenuItem>
                    <MenuItem>
                              <Link
                                        replace={false}
                                        to="/profile/1"
                              >
                                        Profile
                              </Link>
                    </MenuItem>
                    <MenuItem>
                              <Link
                                        replace={false}
                                        to="/settings"
                              >
                                        Settings
                              </Link>
                    </MenuItem>
                    <MenuItem>
                              <a
                                        onClick={[Function]}
                              >
                                        Logout
                              </a>
                    </MenuItem>
          </Menu>
</Adapter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="logo"
>
              Chingu
</div>,
            <Menu
              className=""
              mode="horizontal"
              prefixCls="ant-menu"
              style={
                            Object {
                                          "lineHeight": "64px",
                                        }
              }
              theme="light"
>
              <MenuItem>
                            <Link
                                          replace={false}
                                          to="/"
                            >
                                          Home
                            </Link>
              </MenuItem>
              <MenuItem>
                            <Link
                                          replace={false}
                                          to="/profile/1"
                            >
                                          Profile
                            </Link>
              </MenuItem>
              <MenuItem>
                            <Link
                                          replace={false}
                                          to="/settings"
                            >
                                          Settings
                            </Link>
              </MenuItem>
              <MenuItem>
                            <a
                                          onClick={[Function]}
                            >
                                          Logout
                            </a>
              </MenuItem>
</Menu>,
          ],
          "className": "nav-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Chingu",
              "className": "logo",
            },
            "ref": null,
            "rendered": "Chingu",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <MenuItem>
                  <Link
                                    replace={false}
                                    to="/"
                  >
                                    Home
                  </Link>
</MenuItem>,
                null,
                null,
                <MenuItem>
                  <Link
                                    replace={false}
                                    to="/profile/1"
                  >
                                    Profile
                  </Link>
</MenuItem>,
                <MenuItem>
                  <Link
                                    replace={false}
                                    to="/settings"
                  >
                                    Settings
                  </Link>
</MenuItem>,
                <MenuItem>
                  <a
                                    onClick={[Function]}
                  >
                                    Logout
                  </a>
</MenuItem>,
              ],
              "className": "",
              "mode": "horizontal",
              "prefixCls": "ant-menu",
              "style": Object {
                "lineHeight": "64px",
              },
              "theme": "light",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/"
>
                    Home
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": [Function],
              },
              null,
              null,
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "class",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/profile/1"
>
                    Profile
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Profile",
                    "replace": false,
                    "to": "/profile/1",
                  },
                  "ref": null,
                  "rendered": "Profile",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "class",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/settings"
>
                    Settings
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Settings",
                    "replace": false,
                    "to": "/settings",
                  },
                  "ref": null,
                  "rendered": "Settings",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "class",
                "props": Object {
                  "children": <a
                    onClick={[Function]}
>
                    Logout
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Logout",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": "a",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
